go/host/enclave/guardian.go

```diff
@@ -545,12 +549,52 @@ func (g *Guardian) periodicRollupProduction() {
 			if lastBatchNo.Uint64() > common.L2GenesisSeqNo {
 				fromBatch++
 			}
+
 			producedRollup, err := g.enclaveClient.CreateRollup(fromBatch)
 			if err != nil {
 				g.logger.Error("unable to produce rollup", log.ErrKey, err)
 			} else {
+				fmt.Println(time.Now().String(), " - PRODUCING rollup from config'd rollupInterval - lastBatchNo: ", lastBatchNo)
 				g.sl.L1Publisher().PublishRollup(producedRollup)
+				fmt.Println(time.Now().String(), " - FINISHED PRODUCED rollup from config'd rollupInterval ")
+			}
+
+		case <-rollupCheckTicker.C:
+			if !g.state.IsUpToDate() {
+				// if we're behind the L1, we don't want to produce rollups
+				g.logger.Debug("skipping rollup production because L1 is not up to date", "state", g.state)
+				continue
+			}
+			lastBatchNo, err := g.sl.L1Publisher().FetchLatestSeqNo()
+			if err != nil {
+				g.logger.Warn("encountered error while trying to retrieve latest sequence number", log.ErrKey, err)
+				continue
 			}
+			fmt.Println(time.Now().String(), " - 5sec rollup creation CHECK - lastBatchNo: ", lastBatchNo)
+			fromBatch := lastBatchNo.Uint64()
+			if lastBatchNo.Uint64() > common.L2GenesisSeqNo {
+				fromBatch++
+			}
+			availBatchesSumSize, err := g.enclaveClient.GetBatchesAfterSize(fromBatch)
```

Comment:
Looking at the code now, I think we can simplify this by using the db of the host. 
The host stores all batches as it receives them from the enclave.
Instead of creating this special method in the enclave, we can just interogate the local db

Reply:
Yeah that's a good point, updating it 👍